dynawo-simulation/src/main/java/com/powsybl/dynawo/builders/BuilderEquipmentsList.java

```diff
@@ -48,41 +39,62 @@ public void addEquipments(String[] staticIds, Function<String, T> equipmentsSupp
 
     public void addEquipments(Iterable<String> staticIds, Function<String, T> equipmentsSupplier) {
         staticIds.forEach(id -> addEquipment(id, equipmentsSupplier));
+        reportEmptyList();
+    }
+
+    public void addEquipments(String[] staticIds, Function<String, T> equipmentsSupplier,
+                              EquipmentPredicate<T> equipmentPredicate) {
+        addEquipments(() -> Arrays.stream(staticIds).iterator(), equipmentsSupplier, equipmentPredicate);
+    }
+
+    public void addEquipments(Iterable<String> staticIds, Function<String, T> equipmentsSupplier,
+                              EquipmentPredicate<T> equipmentPredicate) {
+        staticIds.forEach(id -> addEquipment(id, equipmentsSupplier, equipmentPredicate));
+        reportEmptyList();
     }
 
     public void addEquipment(String staticId, Function<String, T> equipmentsSupplier) {
         T equipment = equipmentsSupplier.apply(staticId);
-        if (equipment != null) {
+        if (equipment == null) {
+            handleMissingId(staticId);
+        } else {
+            equipments.add(equipment);
+        }
+    }
+
+    public void addEquipment(String staticId, Function<String, T> equipmentsSupplier,
+                             EquipmentPredicate<T> equipmentPredicate) {
+        T equipment = equipmentsSupplier.apply(staticId);
+        if (equipment == null) {
+            handleMissingId(staticId);
+        } else if (equipmentPredicate.test(equipment, fieldName, reportNode)) {
             equipments.add(equipment);
         } else {
             missingEquipmentIds.add(staticId);
         }
     }
 
-    public boolean checkEquipmentData(ReportNode reportNode) {
+    protected void handleMissingId(String staticId) {
+        missingEquipmentIds.add(staticId);
+        BuilderReports.reportStaticIdUnknown(reportNode, fieldName, staticId, equipmentType);
+    }
+
+    protected void reportEmptyList() {
```

Comment:
Would be more accurate to add a "If" in the name (or to keep the if outside)
```suggestion
    protected void reportIfEmptyList() {
```